go.go

```diff
@@ -259,6 +259,30 @@ const goMarshalField = `{{if eq .Type "bool"}}
 		i++
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if l := len(o.{{.NameTitle}}); l != 0 {
+		buf[i] = {{.Index}}
+		i++
+		x := uint(l)
+		for x >= 0x80 {
+			buf[i] = byte(x | 0x80)
+			x >>= 7
+			i++
+		}
+		buf[i] = byte(x)
+		i++
+		for _, v := range o.{{.NameTitle}} {
+			x1 := uint32((v << 1) ^ (v >> 31))
+			for n := 0; x1 >= 0x80 && n < 8; n++ {
```

Comment:
The `n < 8` is a compiler hint. It should read 4 here though, with 32 bits.

Reply:
Not even sure whether Go still needs it. The compiler has boundary knowledge these days. :-)

Reply:
Dito for the marshal length function below.

Reply:
change this for 32 bits :)